src/generic/maskededit.cpp

```diff
@@ -0,0 +1,1951 @@
+                    lastIsComchars = false;
+                    break;
+                }
+                else
+                {
+                    break; //treat this '{' as a literal
+                }
+            }
+        }
+
+        //Literal treatment
+        if ( isLiteral )
+        {
+            //When previous char was inside a field, finish it.
+            if ( fieldStarted )
+            {
+                fieldsEnd.push_back( pos - 1 ); //Define a new field
+                fieldStarted = false;
+                firstFieldCmm = -1; //reset
+            }
+
+            //'|' means fields ends here, but don't show '|' as any literal
+            if ( ch != '|' )
+            {
+                if ( ch == '\\' )
+                {
+                    i++; //the real literal is next char.
+                    if ( i != mask.end() )
+                    {
+                        ch = *i;
+                    }
+                    else
+                    {
+                        wxFAIL_MSG("Incomplete escaped character");
+                        return false; //'\' at end of mask
+                    }
+                }
+                //Store in the expanded mask. Useful for 'painting' it
+                if ( !wxIsprint(ch) )
+                {
+                    wxFAIL_MSG("Warning: Unprintable character in mask");
+                }
+                maskTmp << ch;
+                maskIDTmp.push_back(mLiteral);
+                sForSize << ch; //This literal has its own size. Don't use 9/M
+                pos++;
+            }
+            i++;
+            lastIsComchars = false;
+        }
+    }
+
+    //If a field is still 'not closed', finish it
+    if ( fieldStarted )
+    {
+        fieldsEnd.push_back( pos - 1 ); //Define a new field
+    }
+
+    if ( fieldsEnd.empty() )
+    {
+        wxFAIL_MSG("The mask must have at least one field");
+        return false;
+    }
+
+    //Copy temporary objects to definitive ones
+    m_ExpMask = maskTmp;
+    m_maskIDs = maskIDTmp;
+    m_arrPosS = fieldsBeg;
+    m_arrPosF = fieldsEnd;
+
+    //Update also at params object
+    m_params.mask = mask; //User definition string, not the expanded
+
+    //Also clear all field dependent parameters.
+    //Changing the mask may change fields. We suppose that previous fields
+    // parameters (choices, allowed/excluded chars, etc) are now invalid.
+    m_params.ResetFields( fieldsEnd.size() );
+
+    //Previous value in the control is lost. Let's show the new empty one
+    m_filled.clear();
+    m_filled.reserve( m_ExpMask.Len() );
+    for (size_t i = 0; i < m_ExpMask.Len(); i++)
+        m_filled.push_back(false);
+
+    SetPlainValue("");
+
+    //Update the minimal size of the control
+    sForSize << 'a'; //give it a tip, for margins
+
+    int extraSize = 0;
+#if wxUSE_COMBOBOX
+    //If our control is a wxComboBox we need extra space for the drop arrow
```

Comment:
Maybe replace the code below with
`extraSize = wxSystemSettings::GetMetric(wxSYS_VSCROLL_ARROW_X, msk_control);`
It would no longer need the include at the top either.